### DISCUSSION TITLE
Where to write flags

### DISCUSSION THREAD

**Discussion by chimney4711** [oobabooga/text-generation-webui#1342] [Q&A] [answered]
Hey guys, sorry if I missed something. Where do I add my flags. Instructions say in the "start" script at run_cmd(..) can't find that. Happy for any hint :) (Edit forgot to mention: Just deleted the old version and downloaded the autoinstaller.zip)

**Comment by BennettFourr** [comment]
If you are useing the auto launcher right click on start-webui.bat and click open with notepad. one it is open you can place your flags affter where it says call python server.py.  

![Screenshot 2023-04-18 181745](https://user-images.githubusercontent.com/94060774/232917528-45fdcf4e-9149-43c5-a4a9-a5fbfc958f34.png)


 If you are manual lanching it just type the flags like this "python server.py --gpu-memory 6 --chat --model-menu --auto-devices --cpu-memory 15GB"

**Reply by chimney4711** [reply]
Thanks, that's how I used to do it as well. However, start-webui.bat is no more. Instead it's called start-windows.bat. The instructions state to add the flags at run_cmd(..) (eg.: run_cmd("python server.py --auto-devices --notebook --model-menu")) but run_cmd... doesn't exist in the script.

**Reply by GGGrim** [reply]
I have the same issue, did you find a solution?

**Reply by MuhannadFM** [reply]
broooo why is none answering

**Reply by Sebastianv650** [reply]
Put them in CMD_FLAGS.txt.

**Comment by BennettFourr** [comment]
![Screenshot 2023-04-19 065352](https://user-images.githubusercontent.com/94060774/233053582-6a340ce2-d333-49e3-8f23-770ab15f2473.png)
try this

**Reply by chimney4711** [reply]
Thanks, that's the first I tried but it throws an error

**Comment by babulee** [comment, marked as answer]
Hello.  You edit webui.py now.
find the following line:
run_cmd("python server.py --auto-devices --chat --model-menu")

**Reply by Leggyweggy** [reply]
sorry to bother you, is this were i can put all of the other available flags now that were previously suppose to append to server.py?

edit: i tried it and it didn't work, neither did replacing the old line with your new one. i'm definitely doing something wrong. could you explain in a little more detail? this seems to be an issue for more than just me

**Reply by chimney4711** [reply]
Thanks a lot, that does the trick! (Sorry for my late response, had eventually figured out)

**Reply by babulee** [reply]
> sorry to bother you, is this were i can put all of the other available flags now that were previously suppose to append to server.py?
> 
> edit: i tried it and it didn't work, neither did replacing the old line with your new one. i'm definitely doing something wrong. could you explain in a little more detail? this seems to be an issue for more than just me

So with the new oobabooga, the launcher is now called start_windows.bat instead of start-webui.bat
In the old oobabooga, you edit start-webui.bat and add your flags after "call python server.py" like "call python server.py --auto-devices --chat"
In the new oobabooga, you do not edit start_windows.bat but edit webui.py, which should be in the root of oobabooga install folder.  You now look for this block of code

def run_model():
    os.chdir("text-generation-webui")
    run_cmd("python server.py --auto-devices --chat --model-menu")

then you append your new flags inside of "python server.py --auto-devices --chat --model-menu"

**Reply by Leggyweggy** [reply]
thanks!

**Reply by Mulakulu** [reply]
Now that there's no longer any webui.py, what do I do now? I see the CMD_FLAGS.txt but it doesn't enable --trust-remote-code when I put that in.

**Comment by guispfilho** [comment]
> Now that there's no longer any webui.py, what do I do now? I see the CMD_FLAGS.txt but it doesn't enable --trust-remote-code when I put that in.

I'm also trying to find out the correct file to edit after the "webui.py" was removed. Did you managed to figure out how to solve this?

**Reply by Mulakulu** [reply]
Yeah, I'm guessing they fixed it or I was mistaken. Maybe I didn't know the format as to how to add flags. Putting "--trust-remote-code" in CMD_GLAGS.txt works now.
![image](https://github.com/oobabooga/text-generation-webui/assets/22845572/e6b15731-fdcc-43cf-93bc-c44059f0048b)
Here's the format I do it in

By the way, I say "it works" since I got the warning "WARNING  trust_remote_code is enabled. This is dangerous." on startup, so I'm guessing it works. I haven't confirmed it through testing.
